c++/src/Writer.cc

```diff
@@ -38,9 +38,10 @@ namespace orc {
     FileVersion fileVersion;
     double dictionaryKeySizeThreshold;
     bool enableIndex;
+    RleVersion rleVersion;
```

Comment:
I think the file version should determine the `RleVersion`. Refer `isNewWriteFormat` and `isDirectV2` on the Java side.

Reply:
To be clear, do we need this `RleVersion` here and in the `WriterOptions`?